packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -0,0 +1,260 @@
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamName });
+  }
+
+  /**
+   * Import an existing delivery stream from its ARN.
+   */
+  static fromDeliveryStreamArn(scope: Construct, id: string, deliveryStreamArn: string): IDeliveryStream {
+    return this.fromDeliveryStreamAttributes(scope, id, { deliveryStreamArn });
+  }
+
+  /**
+   * Import an existing delivery stream from its attributes.
+   */
+  static fromDeliveryStreamAttributes(scope: Construct, id: string, attrs: DeliveryStreamAttributes): IDeliveryStream {
+    if (!attrs.deliveryStreamName && !attrs.deliveryStreamArn) {
+      throw new Error('Either deliveryStreamName or deliveryStreamArn must be provided in DeliveryStreamAttributes');
+    }
+    const deliveryStreamName = attrs.deliveryStreamName ??
+      cdk.Stack.of(scope).splitArn(attrs.deliveryStreamArn!, cdk.ArnFormat.SLASH_RESOURCE_NAME).resourceName;
+
+    if (!deliveryStreamName) {
+      throw new Error(`No delivery stream name found in ARN: '${attrs.deliveryStreamArn}'`);
+    }
+    const deliveryStreamArn = attrs.deliveryStreamArn ?? cdk.Stack.of(scope).formatArn({
+      service: 'firehose',
+      resource: 'deliverystream',
+      resourceName: attrs.deliveryStreamName,
+      arnFormat: cdk.ArnFormat.SLASH_RESOURCE_NAME,
+    });
+    class Import extends DeliveryStreamBase {
+      public readonly deliveryStreamName = deliveryStreamName!;
+      public readonly deliveryStreamArn = deliveryStreamArn;
+      public readonly grantPrincipal = attrs.role ?? new iam.UnknownPrincipal({ resource: this });
+    }
+    return new Import(scope, id);
+  }
+
+  readonly deliveryStreamName: string;
+
+  readonly deliveryStreamArn: string;
+
+  readonly grantPrincipal: iam.IPrincipal;
+
+  constructor(scope: Construct, id: string, props: DeliveryStreamProps) {
+    super(scope, id, {
+      physicalName: props.deliveryStreamName,
+    });
+
+    if (props.destinations.length !== 1) {
+      throw new Error(`Only one destination is allowed per delivery stream, given ${props.destinations.length}`);
+    }
+
+    const role = props.role ?? new iam.Role(this, 'Service Role', {
+      assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+    });
+    this.grantPrincipal = role;
+
+    const destinationConfig = props.destinations[0].bind(this, {});
+
+    const resource = new CfnDeliveryStream(this, 'Resource', {
+      deliveryStreamName: props.deliveryStreamName,
+      deliveryStreamType: 'DirectPut',
+      ...destinationConfig,
+    });
+
+    this.deliveryStreamArn = this.getResourceArnAttribute(resource.attrArn, {
+      service: 'kinesis',
+      resource: 'deliverystream',
+      resourceName: this.physicalName,
+    });
+    this.deliveryStreamName = this.getResourceNameAttribute(resource.ref);
+  }
+}
+
+function setConnections(scope: Construct) {
+  const region = cdk.Stack.of(scope).region;
+  let cidrBlock = RegionInfo.get(region).firehoseCidrBlock;
```

Comment:
Does this work correctly when `region` is `AWS::Region`?

Reply:
will be replaced when #15617 is merged